test/runtests.jl

```diff
@@ -24,21 +24,27 @@ julia = Base.julia_cmd().exec[1]
 end
 
 @testset "build_executable" begin
-    builddir = mktempdir()
     jlfile = joinpath(@__DIR__, "..", "examples", "hello.jl")
-    snoopfile = open(joinpath(builddir, "snoop.jl"), "w") do io
-        write(io, open(read, jlfile))
-        println(io)
-        println(io, "using .Hello; Hello.julia_main(String[])")
+    basedir = mktempdir()
+    relativebuilddir = "build"
+    cd(basedir) do
+        mkdir(relativebuilddir)
+        snoopfile = "snoop.jl"
+        open(snoopfile, "w") do io
+            write(io, open(read, jlfile))
+            println(io)
+            println(io, "using .Hello; Hello.julia_main(String[])")
+        end
+        build_executable(
+            jlfile, snoopfile = snoopfile, builddir = relativebuilddir, verbose = true
+        )
     end
```

Comment:
I think @lucatrv is trying to remove all `cd` from the package. Can you undo that part of the change? You'll just have to have more `joinpath`s. Thanks! :)

Reply:
I think removing `cd`s from the production code is great, but the `cd` here is kind of required to accurately reproduce what at least in my opinion is a pretty standard setup (which triggers the original issue), while avoiding creating files inside PackageCompiler.jl's test directory.